core/jmx/api/src/main/java/org/opennms/netmgt/jmx/JmxUtils.java

```diff
@@ -102,6 +109,24 @@ public static String getGroupName(final Map<String, String> map, final Mbean mbe
         return groupName;
     }
 
+    public static Map<String, String> getRuntimeAttributes(JmxConfigDao jmxConfigDao, String address, Map<String, String> parameters) {
+        Objects.requireNonNull(address);
+        Objects.requireNonNull(parameters);
+        if (jmxConfigDao != null && jmxConfigDao.getConfig() != null) {
+            try {
+                final JmxConnectionConfig config = JmxConnectionConfigBuilder.buildFrom(address, parameters).build();
+                final int port = new JMXServiceURL(config.getUrl()).getPort();
+                final MBeanServer mBeanServer = jmxConfigDao.getConfig().lookupMBeanServer(address, port);
+                if (mBeanServer != null) {
+                    return new HashMap<>(mBeanServer.getParameterMap());
+                }
+            } catch (MalformedURLException ex) {
+                // swallow
```

Comment:
Please print an "Unexpected exception: {}" message here instead of swallowing up the exception.